contrib/cmake/build3rdparty.bash

```diff
@@ -1,17 +1,80 @@
-#!/bin/bash
-# The usual way to use this file is to make a new folder 3rdParty next to your sdk folder, copy this script to it, and run it there.
-# Then building via cmake can refer to these libraries.
-
-git clone https://github.com/Microsoft/vcpkg.git
-cd vcpkg
-./bootstrap-vcpkg.sh
-./vcpkg install --triplet x64-linux zlib
-./vcpkg install --triplet x64-linux cryptopp
-./vcpkg install --triplet x64-linux openssl
-./vcpkg install --triplet x64-linux c-ares
-./vcpkg install --triplet x64-linux curl    
-./vcpkg install --triplet x64-linux sqlite3
-./vcpkg install --triplet x64-linux gtest
-#./vcpkg install --triplet x64-linux libsodium
+# This script is for getting and building the 3rd party libraries that the MEGA SDK uses (some are optional, and some are only needed only by MEGA apps too)
+# 
+# Your 3rdParty library builds should be outside the SDK repo.  We are moving to use vcpkg to build most of them. You can start it like this:
+#
+# mkdir 3rdParty
+# cd 3rdParty
+# git clone https://github.com/Microsoft/vcpkg.git
+# cd vcpkg
+# .\bootstrap-vcpkg.sh -disableMetrics
+#
+# Comment out any libraries that you won't use.
+# If using pdfium, follow the instructions below to get the source code
+#
+# On Mac, go to ports/libraw/CONTROL and comment out the freeglut dependency
+# 
+# From your 3rdParty/vcpkg folder, run this script (in its proper location) with the desired triplet as the parameter.  (usually x64-linux or x64-osx)
 
+export TRIPLET=$1
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+ln -sf $DIR/vcpkg_extra_ports/* ports/
+ln -sf $DIR/vcpkg_extra_triplets/* triplets/
```

Comment:
It seems that `ln` with wildcard does not work on macOS. Tested with two for loops to create the needed links works.

Reply:
Ok, we may need to list those out manually then - those lines are an improvement by @polmr over the prior version by me that just asked the user to copy them manually :-).   There are not many, we can just list them and maintain the list.

Reply:
actually, on consideration, the loop does sound good - could you send me that (or just add directly) - that way it's tested code going into the script?  thanks